extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,146 @@
+
+define([
+    'jquery',
+    './common',
+    './api',
+], function ($, common, api) {
+    "use strict";
+
+
+    /*
+    This function returns the latest version of a package.
+    */
+
+    function checknewversion(updates, packagename, pkgversion) {
+        for (let element of updates) {
+            if (element.name === packagename) {
+                return element.version;
+            }
+        }
+        return pkgversion;
+    }
+
+    /*
+    This function lists down all the updates and requires a user confirmation.
+    */
+
+    async function updatepkg() {
+        $('#updatebtn').unbind();
+        $('#updatebtn').click(async function () {
+            let selectedPackages = common.get_selected_packages();
+            let url = location.href.split('/');
+            let project = 'SWAN_projects/' + url[url.length - 1];
```

Comment:
Why are you getting the project this way, if this information was passed when you clicked the cog??
It's not even correct, since the user can be in a subfolder of the project...